unity-client/Assets/Scripts/MainScripts/DCL/Components/UUIDComponent/OnPointerEventColliders.cs

```diff
@@ -73,6 +76,33 @@ Collider CreateCollider(Renderer renderer)
             return meshCollider;
         }
 
+        private bool AlreadyCreatedColliders(Renderer[] rendererList)
+        {
+            if (colliders == null || colliders.Length == 0)
+                return false;
+
+            if (rendererList.Length != colliders.Length)
+                return false;
+
+            for (int i = 0; i < rendererList.Length; i++)
+            {
+                bool foundChildCollider = false;
+                for (int j = 0; j < colliders.Length; j++)
+                {
+                    if (colliders[j].transform.parent == rendererList[i].transform)
+                    {
+                        foundChildCollider = true;
+                        break;
+                    }
+                }
+
+                if (!foundChildCollider)
+                    return false;
+            }
+            
+            return true;
+        }
```

Comment:
```suggestion
            for (int i = 0; i < rendererList.Length; i++)
            {
                for (int j = 0; j < colliders.Length; j++)
                {
                    if (colliders[j].transform.parent == rendererList[i].transform)
                        return true;
                }
            }
            
            return false;
        }
```

I think this is equivalent and we dont have to rely on extra variables.

Reply:
Actually I think it's not exactly the same, since in this new form we would return true on the first renderer that has a collider matching it as its parent, but we should to check **every renderer and every collider**.

Probably your approach would work but because I think it would be strange to have 1 renderer that is the same as the previous colliders creation and other renderers that are not. But I'd keep it as I pushed it originally just to cover all the renderer and colliders.

If we want to avoid using that bool I could check if the collider's `i` reached the end of the collection without breaking the traversing.

Also this check could be done with LinQ too but I think it would be much less performant.